catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/MatchAnyValidator.java

```diff
@@ -0,0 +1,83 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+
+public class MatchAnyValidator implements AttributeValidator {
+    private final List<AttributeValidator> validators;
+
+    public MatchAnyValidator(List<AttributeValidator> validators) {
+        this.validators = validators;
+    }
+
+    @Override
+    public Optional<AttributeValidationReport> validate(Attribute attribute) {
+        if (attribute == null || CollectionUtils.isEmpty(validators)) {
+            return Optional.empty();
+        }
+
+        List<Optional<AttributeValidationReport>> validationReportList = validators.stream()
+                .map(validator -> validator.validate(attribute))
+                .collect(Collectors.toList());
+
+        return generateValidationReports(validationReportList);
+    }
+
+    private Optional<AttributeValidationReport> generateValidationReports(
+            List<Optional<AttributeValidationReport>> validationReportList) {
+
+        boolean containsValidatorWithoutViolations = false;
+        AttributeValidationReportImpl result = new AttributeValidationReportImpl();
+
+        for (Optional<AttributeValidationReport> attributeValidationReportOptional : validationReportList) {
+
+            if (attributeValidationReportOptional.isPresent()) {
+                AttributeValidationReport attributeValidationReport =
+                        attributeValidationReportOptional.get();
+
+                Set<ValidationViolation> validationViolations =
+                        attributeValidationReport.getAttributeValidationViolations();
+                Set<String> suggestedValues = attributeValidationReport.getSuggestedValues();
+
+
+                if (!containsValidatorWithoutViolations && validationViolations.size() == 0) {
+                    containsValidatorWithoutViolations = true;
+                }
+
+                if (CollectionUtils.isNotEmpty(validationViolations)) {
+                    result.addViolation(validationViolations);
+                }
+
+                if (CollectionUtils.isNotEmpty(suggestedValues)) {
+                    result.addSuggestedValue(suggestedValues);
+                }
+            } else {
+                containsValidatorWithoutViolations = true;
+            }
+        }
+        return containsValidatorWithoutViolations ? Optional.empty() : Optional.of(result);
```

Comment:
Based on this line, you could short-circuit processing in the loop above. Something like the following should work:
```java
    private Optional<AttributeValidationReport> generateValidationReports(
            List<Optional<AttributeValidationReport>> validationReportList) {

        AttributeValidationReportImpl result = new AttributeValidationReportImpl();

        for (Optional<AttributeValidationReport> attributeValidationReportOptional : validationReportList) {

            if (attributeValidationReportOptional.isPresent()) {
                AttributeValidationReport attributeValidationReport =
                        attributeValidationReportOptional.get();

                Set<ValidationViolation> validationViolations =
                        attributeValidationReport.getAttributeValidationViolations();
                Set<String> suggestedValues = attributeValidationReport.getSuggestedValues();


                if (CollectionUtils.isEmpty(validationViolations)) {
                    return Optional.empty();
                }

                result.addViolation(validationViolations);

                if (CollectionUtils.isNotEmpty(suggestedValues)) {
                    result.addSuggestedValue(suggestedValues);
                }
            } else {
                return Optional.empty();
            }
        }
        return Optional.of(result);
    }
```